### ISSUE TITLE
so-bogus @ 046dc10 in bogus-interface/extern/ cannot be attached

### ISSUE THREAD

**Issue by lanchen2019** [lijieumn/bogus-interface#1]
Hi Jie，
When I compile the code, I met the following problem.
![image](https://user-images.githubusercontent.com/46516710/67016567-f900a400-f12a-11e9-84b6-a3ab55bc760b.png)
I think the reason is the submodule(so-bogus @ 046dc10) in bogus-interface/extern/ cannot be attached. 
Could you please update the codes? I can't wait to re-simulate your wonderful results. 
Thank you very much!

**Comment by lijieumn** [comment]
Hi there,
When you set up the code, have you run the following command to obtain all the submodules?
```bash
git submodule update --init --recursive
```
This command will enables you to download all necessary dependencies. For more details about installing the code, you can refer to the [installation page](https://github.com/lijieumn/argus-distribution/blob/master/INSTALL.md).

Feel free to let me know if you have further questions.

Jie

**Comment by zzheng90** [comment]
Excuse me, i have the same problem above, **git submodule update --init --recursive** did not work, the two submodules documents 'bogus-interface' and 'libarcsim' are empty inside.

**Comment by lijieumn** [comment]
> Excuse me, i have the same problem above, **git submodule update --init --recursive** did not work, the two submodules documents 'bogus-interface' and 'libarcsim' are empty inside.

Hi, the submodules 'bogus-interface' and 'libarcsim' are both under the project [Argus-distribution](https://github.com/lijieumn/argus-distribution). You might have to download that project and run `git submodule update --init --recursive` there. Let me know if you have further questions.

Jie

**Comment by zzheng90** [comment]
In fact, the Extern file is empty even after running the command 'git submodule update --init --recursive'.

**Comment by zzheng90** [comment]
I did like this:
1. downmoad the 'argus-distribution-master.zip'
2. unzip the document
3. In the document root directory, tab command: git init
4. then tab command: git submodule update --init --recursive
But the two submodules are still empty inside: 
deps/bogus-interface 
deps/libarcsim
Could you tell me where is wrong on my side?

**Comment by lijieumn** [comment]
> I did like this:
> 
> 1. downmoad the 'argus-distribution-master.zip'
> 2. unzip the document
> 3. In the document root directory, tab command: git init
> 4. then tab command: git submodule update --init --recursive
>    But the two submodules are still empty inside:
>    deps/bogus-interface
>    deps/libarcsim
>    Could you tell me where is wrong on my side?

When you download the Zip file directly, the .git information seems missing from the directory. I suggest trying to download the project by the following command

```
git clone https://github.com/lijieumn/argus-distribution.git
```

and then go to the directory and run the following command to update the submodules

```
git submodule update --init --recursive
```

**Comment by zzheng90** [comment]
Yes, your are right, i arrived to update the submodules by your method, but when I tab command: make -j, 
I get another problem as follows, could you tell why? 
![image](https://user-images.githubusercontent.com/73642409/111653524-498c9d00-8808-11eb-9402-6fa7be078c1e.png)
![image](https://user-images.githubusercontent.com/73642409/111654098-bacc5000-8808-11eb-85dd-aa301d199942.png)

**Comment by zzheng90** [comment]
I dont know if others have also this kind of problem or not? But I realy appreciate your help.
